File: modules/calc.py

class calculate:

	def __init__(self, age:int, ldl:int, cholesterol:int, hdl:int, diastolic:int, systolic:int, diabetes:bool, smoker:bool):

		self.age = age
		self.ldl = ldl
		self.cholesterol = cholesterol
		self.hdl = hdl
		self.diastolic = diastolic
		self.systolic = systolic
		self.diabetes = diabetes
		self.smoker = smoker

		self.risk = False
		self.comparative = False

	def age_risk(self):

		if 30 <= self.age <= 34:
			self.comparative = 3
			return -1
		elif 35 <= self.age <= 39:
			self.comparative = 5
			return 0
		elif 40 <= self.age <= 44:
			self.comparative = 7
			return 1
		elif 45 <= self.age <= 49:
			self.comparative = 11
			return 2
		elif 50 <= self.age <= 54:
			self.comparative = 14
			return 3
		elif 55 <= self.age <= 59:
			self.comparative = 16
			return 4
		elif 60 <= self.age <= 64:
			self.comparative = 21
			return 5
		elif 65 <= self.age <= 69:
			self.comparative = 25
			return 6
		elif 70 <= self.age <= 74:
			self.comparative = 30
			return 7

	def ldl_risk(self):

		if self.ldl < 100:
			return -3
		elif 100 <= self.ldl <= 129:
			return 0 
		elif 130 <= self.ldl <= 159:
			return 0 
		elif 160 <= self.ldl < 190:
			return 1
		else:
			return 2

	def cholesterol_risk(self):

		if self.cholesterol < 160:
			return -3
		elif 160 <= self.cholesterol <= 199:
			return 0 
		elif 200 <= self.cholesterol <= 239:
			return 1
		elif 240 <= self.cholesterol < 279:
			return 2
		else:
			return 3

	def hdl_risk(self):

		if self.hdl < 35:
			return 2
		elif 35 <= self.hdl <= 44:
			return 1
		elif 45 <= self.hdl <= 49:
			return 0
		elif 50 <= self.hdl < 59:
			return 0
		else:
			return -1

	def pressure_risk(self):

		if self.diastolic < 80:
			if self.systolic < 130:
				return 0
			elif 130 <= self.systolic <= 139:
				return 1
			elif 140 <= self.systolic <= 159:
				return 2
			else:
				return 3

		elif 80 <= self.diastolic <= 84:
			if self.systolic < 130:
				return 0
			elif 130 <= self.systolic <= 139:
				return 1
			elif 140 <= self.systolic <= 159:
				return 2
			else:
				return 3

		elif 85 <= self.diastolic <= 89:
			if self.systolic < 140:
				return 1
			elif 140 <= self.systolic <= 159:
				return 2
			else:
				return 3

		elif 90 <= self.diastolic <= 99:
			if self.systolic < 160:
				return 2
			else:
				return 3

		else:
			return 3

	def diabetes_risk(self):

		if self.diabetes:
			return 2
		else:
			return 0

	def smoker_risk(self):

		if self.smoker:
			return 2
		else:
			return 0

	def calculate_risk(self):

		points = self.age_risk() + self.ldl_risk() + self.cholesterol_risk() + self.hdl_risk() + self.pressure_risk() + self.diabetes_risk() + self.smoker_risk()

		if points == -3:
			self.risk = 1
		elif points == -2:
			self.risk = 2
		elif points == -1:
			self.risk = 2
		elif points == 0:
			self.risk = 3
		elif points == 1:
			self.risk = 4
		elif points == 2:
			self.risk = 4
		elif points == 3:
			self.risk = 5
		elif points == 4:
			self.risk = 7
		elif points == 5:
			self.risk = 9
		elif points == 6:
			self.risk = 11
		elif points == 7:
			self.risk = 14
		elif points == 8:
			self.risk = 18
		elif points == 9:
			self.risk = 22
		elif points == 10:
			self.risk = 27
		elif points == 11:
			self.risk = 33
		elif points == 12:
			self.risk = 40
		elif points == 13:
			self.risk = 47
		else:
			self.risk = 56

		print(f"10 year CHD risk: {self.risk}%")
		print(f"Average for {self.age}: {self.comparative}")

		return {"risk" : self.risk, "average" : self.comparative}


obj = calculate(
	age = 66, # 6
	ldl = 160, #
	cholesterol = 101,
	hdl = 44,
	diastolic = 90,
	systolic = 133,
	diabetes = True,
	smoker = False
)

obj.calculate_risk()

File: modules/calc.js

var NWS = NWS || function () {
    return {
        initNamespace: function (namespace, f) {
            if (typeof (f) === 'undefined') {
                alert('Must define a class');
                return;
            }

            var o = this, i, j, d;
            d = namespace.split(".");
            for (j = (d[0] == "NWS") ? 1 : 0; j < d.length; ++j) {
                o[d[j]] = o[d[j]] || (j == d.length - 1 ? f() : {});
                o = o[d[j]];
            }

            return o;
        }
    };
}();

            /*

            JavaScript:  Framingham Heart Disease Risk Calculator
            February 28, 1999

            This calculator was created by Charles Hu for the Medical College of
            Wisconsin General Internal Medicine Clinic.  This calculator may not be
            copied without consent from the author.  Please contact him at
            chuckhu@hotmail.com

            Modified (heavily) 29-Oct-2015 by Northwoods.  Original yielding incorrect results.
            (Though possibly correct for the pre-2008 Framingham updates).

            data taken from: http://www.nhlbi.nih.gov/health-pro/guidelines/current/cholesterol-guidelines/quick-desk-reference-html/10-year-risk-framingham-table#women


            */

            //


            NWS.initNamespace("NWS.Calculators", function () {
                // [sex][age index]
                var agePtsArray = [
                [-9, -4, 0, 3, 6, 8, 10, 11, 12, 13],
                [-7, -3, 0, 3, 6, 8, 10, 12, 14, 16]
                ];
                var getAgeIndex = function (age) {
                    if (age < 35) return 0;
                    if (age < 40) return 1;
                    if (age < 45) return 2;
                    if (age < 50) return 3;
                    if (age < 55) return 4;
                    if (age < 60) return 5;
                    if (age < 65) return 6;
                    if (age < 70) return 7;
                    if (age < 75) return 8;
                    return 9;
                };


                // [sex][ageIndex][cholesterol index]
                var totCholPtsArray = [
                [
                [0, 4, 7, 9, 11],
                [0, 4, 7, 9, 11],
                [0, 3, 5, 6, 8],
                [0, 3, 5, 6, 8],
                [0, 2, 3, 4, 5],
                [0, 2, 3, 4, 5],
                [0, 1, 1, 2, 3],
                [0, 1, 1, 2, 3],
                [0, 0, 0, 1, 1],
                [0, 0, 0, 1, 1]
                ],
                [
                [0, 4, 8, 11, 13],
                [0, 4, 8, 11, 13],
                [0, 3, 6, 8, 10],
                [0, 3, 6, 8, 10],
                [0, 2, 4, 5, 7],
                [0, 2, 4, 5, 7],
                [0, 1, 2, 3, 4],
                [0, 1, 2, 3, 4],
                [0, 1, 1, 2, 2],
                [0, 1, 1, 2, 2]
                ]
                ];
                var getTotCholIndex = function (totChol) {
                    if (totChol < 160) return 0;
                    if (totChol < 200) return 1;
                    if (totChol < 240) return 2;
                    if (totChol < 280) return 3;
                    return 4;
                };

                // [sex][age index][smoker (n/y)]
                var smokingPtsArray = [
                [
                [0, 8],
                [0, 8],
                [0, 5],
                [0, 5],
                [0, 3],
                [0, 3],
                [0, 1],
                [0, 1],
                [0, 1],
                [0, 1]
                ], [
                [0, 9],
                [0, 9],
                [0, 7],
                [0, 7],
                [0, 4],
                [0, 4],
                [0, 2],
                [0, 2],
                [0, 1],
                [0, 1]
                ]
                ];

                //[hdl index].  sex/age irrelevant.
                // for a 1-d array, this is overkill, but it preserves the pattern.
                var hdlPtsArray = [-1, 0, 1, 2]
                var getHdlIndex = function (hdl) {
                    if (hdl > 59) return 0;
                    if (hdl > 49) return 1;
                    if (hdl >= 40) return 2;
                    return 3;
                };

                //[sex][bp index][untreated/treated].  age irrelevant.
                var bpPtsArray = [
                [
                [0, 0],
                [0, 1],
                [1, 2],
                [1, 2],
                [2, 3]
                ],
                [
                [0, 0],
                [1, 3],
                [2, 4],
                [3, 5],
                [4, 6]
                ]
                ]
                var getBpIndex = function (bp) {
                    if (bp < 120) return 0;
                    if (bp < 130) return 1;
                    if (bp < 140) return 2;
                    if (bp < 160) return 3;
                    return 4;
                };

                //[sex][risk]
                var tenYearRisk = [
                [1, 1, 1, 1, 1, 2, 2, 3, 4, 5, 6, 8, 10, 12, 16, 20, 25],
                [1, 1, 1, 1, 2, 2, 3, 4, 5, 6, 8, 11, 14, 17, 22, 27]
                ];
                var getTenYearRisk = function (sexIndex, points) {
                    var pointsIndex = points;
                    if (sexIndex == 0) {
                        if (points < 0) return '<1';
                        if (points >= 17) return '>=30';
                    }
                    else {
                        if (points < 9) return '<1';
                        if (points >= 25) return '>=30';
                        pointsIndex = points - 9;
                    }
                    return tenYearRisk[sexIndex][pointsIndex];
                };


                var checkSex = function (form) {
                    if (form.sex[0].checked) return 0;
                    if (form.sex[1].checked) return 1;
                    alert("Please select your sex.");
                    return -1
                };

                var checkAge = function (form) {
                    var age = form.age.value;
                    if (!checkNum(age)) {
                        alert("Please enter your age.");
                        form.age.select();
                        form.age.focus();
                        return -1;
                    }
                    if ((age < 20) || (age > 79)) {
                        if (!confirm("Results are ONLY applicable between the ages of 20 and 79.  Do you still want to proceed?")) {
                            form.age.select();
                            form.age.focus();
                            clearForm(form);
                            return -1;
                        }
                    }
                    return getAgeIndex(age);
                };

                var checkSmoker = function (form) {
                    if (form.smoker[1].checked) return 0;
                    if (form.smoker[0].checked) return 1;
                    alert("Do you smoke?");
                    return -1;
                };

                var checkBp = function (form, complete) {
                    var bp = form.bp.value;
                    if (!checkNum(bp)) {
                        if (complete == 1) {
                            alert("Please enter your systolic blood pressure.");
                            form.bp.focus();
                            form.bp.select();
                            return -1
                        }
                        form.bp.value = '';
                        return 0; // bp index of 0 always assigns 0 points
                    }
                    return getBpIndex(bp);
                };

                var checkBpTreated = function (bpIndex, form, complete) {
                    if (form.bpTreated[1].checked) return 0;
                    if (form.bpTreated[0].checked) return 1;
                    if (bpIndex == 0) return 0; // low bp doesn't require treatment
                    alert("Is your blood pressure being treated?");
                    return -1;
                };

                var checkTotChol = function (form, complete) {
                    var totChol = form.totChol.value;
                    if (!checkNum(totChol)) {
                        if (complete == 1) {
                            alert("Please enter your total cholesterol.");
                            form.totChol.focus();
                            form.totChol.select();
                            return -1
                        }
                        form.totChol.value = '';
                        return 0; // totChol index of 0 always assigns 0 points
                    }
                    return getTotCholIndex(totChol);
                };

                var checkHdl = function (form, complete) {
                    var hdl = form.hdl.value;
                    if (!checkNum(hdl)) {
                        if (complete == 1) {
                            alert("Please enter your total cholesterol.");
                            form.hdl.focus();
                            form.hdl.select();
                            return -1
                        }
                        form.hdl.value = '';
                        return 1; // hdl index of 1 always assigns 0 points
                    }
                    return getHdlIndex(hdl);
                };

                var checkNum = function (val) {
                    if ((val == null) || (isNaN(val)) || (val == "") || (val < 0)) {
                        return false
                    }
                    return true;
                };

                var isNumeric = function (val) {
                    if ((val == null) || (isNaN(val)) || (val == "")) {
                        return false
                    }
                    return true;
                };


                var _public = {};

                _public.riskCalc = function (form, complete) {

                    var sexIndex = checkSex(form);
                    if (sexIndex == -1) return false;

                    var ageIndex = checkAge(form);
                    if (ageIndex == -1) return false;

                    var smokerIndex = checkSmoker(form);
                    if (smokerIndex == -1) return false;

                    var bpIndex = checkBp(form, complete);
                    if (bpIndex == -1) return false;

                    var bpTreatedIndex = checkBpTreated(bpIndex, form);
                    if (bpTreatedIndex == -1) return false;

                    var totCholIndex = checkTotChol(form, complete);
                    if (totCholIndex == -1) return false;

                    var hdlIndex = checkHdl(form, complete);
                    if (hdlIndex == -1) return false;

                    var agePts = agePtsArray[sexIndex][ageIndex];
                    var smokerPts = smokingPtsArray[sexIndex][ageIndex][smokerIndex];
                    var bpPoints = bpPtsArray[sexIndex][bpIndex][bpTreatedIndex];
                    var totCholPts = totCholPtsArray[sexIndex][ageIndex][totCholIndex];
                    var hdlPts = hdlPtsArray[hdlIndex];

                    var totalpts = agePts + smokerPts + bpPoints + totCholPts + hdlPts;
                    var tenYearRiskValue = getTenYearRisk(sexIndex, totalpts);

                    document.getElementById("agePoints").innerHTML = agePts;
                    document.getElementById("smokerPoints").innerHTML = smokerPts;
                    document.getElementById("bpPoints").innerHTML = bpPoints;
                    document.getElementById("totCholPoints").innerHTML = totCholPts;
                    document.getElementById("hdlPoints").innerHTML = hdlPts;
                    document.getElementById("totalPts").innerHTML = totalpts + '.';
                    document.getElementById("tenYearRisk").innerHTML = '(' + tenYearRiskValue + '% risk of heart disease in 10 years.)';

                    document.getElementById("results").style.display = '';

                    // for testing only - uncomment to directly test total points to total risk lookup
                    //var ptsForTest = form.pointsTestInput.value;
                    //if (isNumeric(ptsForTest)) {
                    //    var tenYearRiskValue = getTenYearRisk(sexIndex, ptsForTest);
                    //    document.getElementById("pointsTestRisk").innerHTML = tenYearRiskValue;
                    //}
                };

                _public.completeReset = function (form) {
                    form.reset();
                    document.getElementById("results").style.display = 'none';
                    var pointsElements = document.getElementsByClassName("points");
                    for (var i = 0; i < pointsElements.length; i++)
                        pointsElements[i].innerHTML = '';
                };

                return _public;

            });

